app/controllers/organizations/pets_controller.rb

```diff
@@ -21,6 +21,13 @@ def edit
 
   def show
     @active_tab = determine_active_tab
+    return unless @active_tab == "tasks"
+
+    @task = params[:task_id] ? @pet.tasks.find(params[:task_id]) : @pet.tasks.build
+    respond_to do |format|
+      format.html
+      format.turbo_stream
+    end
```

Comment:
I think we can get rid of this task logic from the PetsController now. Let me know if I am forgetting something.

Reply:
Hmm, hold off on this comment. I think I may have different thoughts now.

I meant to get more review progress on this tonight, but I ended up falling a little ill after the meeting. Hopefully, I will feel better tomorrow and finish it up then. I think I may have a fix for the delay you noticed.

Reply:
I originally did remove it but it must have slipped back in with a merge on accident. As of right now it does nothing but I'm curious on you fix for the visual delay.

Reply:
For some reason I thought I removed this but I guess not. I will have to rework a few things. The edit form cancel is all going through the tasks_controller but the new form cancel was still relying on this.

I'm looking forward to your thoughts on the delay.

Reply:
Last night, I was thinking the delay behavior must have to do with lazy loading or instant links. Poking around, I haven't quite figured out how to eager load the tasks though with the tab. However, I'm not sure we really have to worry about it. Right now, your solution is just loading the tasks in a separate request from the tab 'template', which is fine since I don't see any n+1 issues.

Reply:
> Last night, I was thinking the delay behavior must have to do with lazy loading or instant links. Poking around, I haven't quite figured out how to eager load the tasks though with the tab. However, I'm not sure we really have to worry about it. Right now, your solution is just loading the tasks in a separate request from the tab 'template', which is fine since I don't see any n+1 issues.

My first thought was lazy/eager load but couldn't figure anything obvious out. I agree I don't thing the delay is a huge issue right now. I was thinking maybe in a future revision each tab would be updated with turbo instead of the whole page re-rendering, which I think would possibly help (or maybe not haha). Just a thought.